rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -2517,6 +2537,14 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
         }
 
         execution.scheduledExecution=se
+
+        // If there is a preExecutionCheckError, put that error into the execution's extraMetadataMap.
+        if(!beforeExecutionResult.isSuccessful() && beforeExecutionResult.isTriggeredByJobEvent(JobPreExecutionEvent)) {
+            Map<String, String> extraMeta = new HashMap<>()
+            extraMeta.put(JobPreExecutionEvent.getName(), beforeExecutionResult.getErrorMessage())
+            execution.setExtraMetadataMap(extraMeta)
```

Comment:
this would overwrite any other extra metadata, I think it should merge with any existing Map data instead

Reply:
Good point. I will make it a merge operation. Thanks.